src/main/java/com/sparrowwallet/sparrow/io/satochip/SecureChannelSession.java

```diff
@@ -0,0 +1,245 @@
+package com.sparrowwallet.sparrow.io.satochip;
+
+import com.sparrowwallet.drongo.crypto.AESKeyCrypter;
+
+import com.sparrowwallet.drongo.bip47.SecretPoint;
+import com.sparrowwallet.drongo.crypto.ECKey;
+import com.sparrowwallet.drongo.crypto.Key;
+import com.sparrowwallet.drongo.crypto.EncryptedData;
+
+import com.sparrowwallet.drongo.Utils;
+
+import javax.crypto.Mac;
+import javax.crypto.Cipher;
+import javax.crypto.KeyAgreement;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.IOException;
+import java.security.*; // todo remove?
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.nio.ByteBuffer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles a SecureChannel session with the card.
+ */
+public class SecureChannelSession {
```

Comment:
Prefer `static final`. Multiple constants are unused. Multiple fields should be declared as local variable instead of fields.